packages/core/metrics/core-metrics-collectors-server-internal/src/cgroup/gather_info.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import fs from 'fs/promises';
+
+import { GROUP_CPU, GROUP_CPUACCT } from './constants';
+
+const CONTROL_GROUP_RE = new RegExp('\\d+:([^:]+):(/.*)');
+const CONTROLLER_SEPARATOR_RE = ',';
+const PROC_SELF_CGROUP_FILE = '/proc/self/cgroup';
+
+/**
+ * Check whether the contents of /proc/self/cgroup indicate that we are running in a cgroup v2
+ *
+ * @note cgroup v2 is always in the format "0::<PATH>". See https://www.kernel.org/doc/Documentation/cgroup-v2.txt.
+ */
+function isCgroups2(procSelfLines: string[]): boolean {
+  return procSelfLines.length === 1 && procSelfLines[0].trim().startsWith('0::');
+}
+
+async function readProcSelf(): Promise<string[]> {
+  const data = (await fs.readFile(PROC_SELF_CGROUP_FILE)).toString();
+  return data.split(/\n/).filter((line) => line.trim().length > 0);
+}
+
+type Result = [data: Record<string, string>, v2: boolean];
```

Comment:
NIT: why using a fixed-length list type here?

```ts
interface Result {
  data: Record<string, string>;
  v2: boolean;
}
```

Reply:
No strong reason, just thought it was a bit more terse. With labeled entries you still get the "self documentation" of objects too. I'll use an object in this instance as there's no good reason not to.